src/Compilers/VisualBasic/Test/Semantic/Compilation/CompilationAPITests.vb

```diff
@@ -1734,6 +1735,118 @@ BC2014: the value '_' is invalid for option 'RootNamespace'
+        <InlineData(WellKnownMemberNames.ExclusiveOrOperatorName, "Public Shared Operator Xor(left As Integer, right As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.BitwiseOrOperatorName, "Public Shared Operator Or(left As Integer, right As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.BitwiseAndOperatorName, "Public Shared Operator And(left As Integer, right As Integer) As Integer")>
+        Public Sub CreateBuiltinBinaryOperator_Supported(name As String, display As String)
+            Dim compilation = CreateCompilation("")
+            Dim intType = compilation.GetSpecialType(SpecialType.System_Int32)
+            Dim op = compilation.CreateBuiltinOperator(name, intType, intType, intType, isChecked:=False)
+            Dim result = op.ToDisplayString()
+            AssertEx.Equal(display, result)
+        End Sub
+
+        <Fact>
+        Public Sub CreateBuiltinBinaryOperator_NotSupported()
+            Dim compilation = CreateCompilation("")
+            Dim intType = compilation.GetSpecialType(SpecialType.System_Int32)
+
+            ' c# binary operator name
+            Assert.Throws(Of ArgumentException)(Function() compilation.CreateBuiltinOperator(WellKnownMemberNames.IncrementOperatorName, intType, intType, intType, isChecked:=False))
+
+            ' unary operator name
+            Assert.Throws(Of ArgumentException)(Function() compilation.CreateBuiltinOperator(WellKnownMemberNames.UnaryPlusOperatorName, intType, intType, intType, isChecked:=False))
+        End Sub
+
+        <Fact>
+        Public Sub CreateBuiltinBinaryOperator_NullChecks()
+            Dim compilation = CreateCompilation("")
+            Dim intType = compilation.GetSpecialType(SpecialType.System_Int32)
+            Assert.Throws(Of ArgumentNullException)(Function() compilation.CreateBuiltinOperator(WellKnownMemberNames.AdditionOperatorName, Nothing, intType, intType, isChecked:=False))
+            Assert.Throws(Of ArgumentNullException)(Function() compilation.CreateBuiltinOperator(WellKnownMemberNames.AdditionOperatorName, intType, Nothing, intType, isChecked:=False))
+            Assert.Throws(Of ArgumentNullException)(Function() compilation.CreateBuiltinOperator(WellKnownMemberNames.AdditionOperatorName, intType, intType, Nothing, isChecked:=False))
+        End Sub
+
+        <Theory>
+        <InlineData(WellKnownMemberNames.UnaryPlusOperatorName, "Public Shared Operator +(value As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.UnaryNegationOperatorName, "Public Shared Operator -(value As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.OnesComplementOperatorName, "Public Shared Operator Not(value As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.ImplicitConversionName, "Public Shared Operator CType(value As Integer) As Integer")>
+        <InlineData(WellKnownMemberNames.ExplicitConversionName, "Public Shared Operator CType(value As Integer) As Integer")>
```

Comment:
I don't see C# tests for these. I also think they won't work since `UnaryOperatorNameFromSyntaxKindIfAny` doesn't seem to handle them. Not sure if this is by design?

Reply:
c# doesn't support these afaict.  i could add tests, but just that they fail.

Reply:
> c# doesn't support these afaict

I thought these are:

```csharp
public class C
{
    public static implicit operator int(C c) => 0;
}
```

So I don't quite understand why they shouldn't work in C# but work in VB.
I understand that these are different languages and it's not always a one-to-one mapping, but I don't see the reason for this specific case.

Reply:
afaict, tehse are not the same in the lnaguages, and c# doesn't represent things in that fashion.  You can see how it does it with how c# generates these SynthesizedIntrinsic operators currently.